src/main/java/hudson/plugins/emailext/plugins/content/FailedTestsContent.java

```diff
@@ -64,21 +65,22 @@ public boolean acceptsMacroName(String macroName) {
 
 
     @Override
-    public String evaluate(AbstractBuild<?, ?> build, TaskListener listener, String macroName) {
+    public String evaluate(AbstractBuild<?, ?> build, TaskListener listener, String macroName) throws MacroEvaluationException{
         return evaluate(build, build.getWorkspace(), listener, macroName);
     }
 
     @Override
-    public String evaluate(Run<?, ?> run, FilePath workspace, TaskListener listener, String macroName) {
+    public String evaluate(Run<?, ?> run, FilePath workspace, TaskListener listener, String macroName) throws MacroEvaluationException {
         AbstractTestResultAction<?> testResult = run.getAction(AbstractTestResultAction.class);
         SummarizedTestResult result = prepareSummarizedTestResult(testResult);
 
         switch (outputFormat) {
             case "yaml":
                 try {
                     return result.toYamlString();
-                } catch (Exception e) { }
-                return "Bad format";
+                } catch (Exception e) {
+                    throw new MacroEvaluationException("Bad format", MACRO_NAME, e.getCause());

```

Comment:
I have two comments here:
- The message `"Bad format"` is not really helpful at all. I don't think the occurrence of the exception is because of user mistake. On the contrary, this exception indicates either a bug in Jackson or our own intermediate classes. I will leave it to you to come up with a better message.
- I don't see any benefit of using the `e.getCause()` as the cause of the new exception. In fact, this goes on the same lines as the above. If the cause of the exception is a bug in local code (not Jackson), then that will get swallowed and it will then be a nightmare to debug it.

Reply:
If we catch only `JsonProcessingException` as you suggested and show the cause, it would make sense. Having a cascade of `catch` clauses will help throwing the correct exception i think.